### ISSUE TITLE
Guide help

### ISSUE THREAD

**Issue by StarkZhou** [RaymanNg/3D-Wind-Field#29]
I have a question about the wind ,that if i modify the shader,can i draw the profile wind?
this project is draw a horizontal wind,and can select the height,if i chang the code to draw a profile wind in a lon or lat ,would it can make it in Cesium.
last question ,my eventual aim is to draw a cloud in cesium which can fly dynamic due to the data of nc of wind which is wrf output data,Is it possiblely in theory？

the question which control the range of wind is solved by modify the shader in your code by myself.
my english is poor ,please tell me the answer. thank you very much.

**Comment by RaymanNg** [comment]
What do you mean by "profile wind" ?

If you want to use different equation to render the wind, you can modify the GLSL code in getWind.frag

If you want to draw something else like cloud, try to make use of Cesium API if possible because it is much easier than write a shader

**Comment by StarkZhou** [comment]
Sorry，my question is not clear.It means section wind field, i.e. vertical structure of wind. I have  23
lev data and i need to draw wind at a fixed latitude or longitude to analyse the vertical structure of the wind. I have read your code many times and I'm going to modify it to meet my requirements. 
 Can i make it based on your code?

Beside,i also want to draw cloud by cesium with particlesStsyem.The particles flow according to the wind data. I'm currently going to do this by writing shaders with Cesium. If you have experience in cloud simulation, can you say something about it. 

Sorry again about my unclear question. Thanks again.

**Comment by RaymanNg** [comment]
> Can i make it based on your code?

Yes, of course you can.

> i also want to draw cloud

Sorry I don't know how to render cloud, and I don't think you can easily draw cloud with particlesStsyem class, because cloud has complex shape, it is not a simple geometry. Again I recommend you to make use of Cesium API if its performance is acceptable.

**Comment by StarkZhou** [comment]
thank you.the cloud simulation is later work and i will use Cesium Api first according to your advice.And now,I'm trying to draw  section wind.
I have another questions to ask.  Can i changed the level component as height component.which means height substitution level. i set a fixed lat for example,and modefiy the computing.js ,but i do not know how to modify the particleRender.js and the shaders associated.
I don't know what the logic is, that is, the rendering process. I hope you can give me the guide. 
thanks again   : )

**Comment by RaymanNg** [comment]
You can replace the "lev" dimension with your height data by editing the NetCDF file (read the README for more information)

"CustomPrimitive" class encapsulates the GLSL program, you can find the GLSL code you need to modify, and then check if the relevant CustomPrimitive object needs modification.

I think you can just focus on particleComputing.js if the edit of NetCDF data is the only change you make.

**Comment by StarkZhou** [comment]
I have changed particleComputing.js and the relevant CustomPrimitive. now,i want to know the process in the particleRendering.js. 

In the particleRendering.js, i see that the function "createSegmentsGeometry”, which creat a geometry with "st" and "normal" array.I have question about it. 

First:In my comprehend,"st" is normalized textureCoord, what does the repeatVertex do in the "for" loops. Is it represent per particle Texture is a rectangle with four vertex?  the code is 
for (var i = 0; i < repeatVertex; i++) {
                    st.push(s / userInput.particlesTextureSize);
                    st.push(t / userInput.particlesTextureSize);
                }
 Second: I really don't konw the mean of  "normal"  array which your put pointToUse and offsetSign to. And i read the segmentDraw.vert , find the pointToUse is just to judge the currentProjectedCoord or nextProjectedCoord by select  [-1 ,1].Can you explain the logic in it.  The "vec4 offset "which  in the segmentDraw.vert located in Line 87,used to do what works? I do not know the "offsetSign" which  in the segmentDraw.vert located in Line 85  neither.

Third:In the particleRendering.js ,you creat three CustomPrimitive,segment ,trail,and sceen. I guss if the segment is to creat the per one line in one frame , then the trail.primitive is make trailsColor strictly decreased. at last ,In the Screen.primitive,draw them all;

Sorry for many question,I hope you will give me some advice。 If you have any another Contact adress，i also want send the code which modefiy based on your code to you to find any mistakes. of course In github is ok.

Thanks you again and give you  my best wishes,hope your detailed guidance.

**Comment by RaymanNg** [comment]
Explanation:
1. "st" is used to fetch all the points of the line, which stored in a texture
2. The reason why I used "normal" as variable name is the restriction of Cesium shader helper classes. Actually the "normal" variable in segmentDraw.vert is not normal.
3. I used the algorithm explained in this [blog post](https://mattdesl.svbtle.com/drawing-lines-is-hard#triangulated-lines_1). I stretch the line by duplicating the vertices and move them along normal direction, one is `point + normal` and the other is `point - normal`. 
P.S. this [blog post](http://www.danielkoitzsch.de/blog/2015/09/08/render-thick-lines-in-opengl/) may help you understand what I said.
4. Yes, the segment will fade out after some time
